bench_method.go

```diff
@@ -39,6 +40,9 @@ type peerTransport struct {
 type benchmarkMethod struct {
 	serializer encoding.Serializer
 	req        *transport.Request
+
+	streamRequest         *transport.StreamRequest
+	streamRequestMessages [][]byte
```

Comment:
Hum, `benchmarkMethod` is now responsible of doing 2 things, unary request and streaming request. Is there any chance to decouple this by using an interface? For instance we could have:

```
type benchmarker interface {
WarmTransports(n int, tOpts TransportOptions, resolved resolvedProtocolEncoding, warmupRequests int) ([]peerTransport, error)
Call(t transport.Transport) (time.Duration, error)
}
```

and `type struct benchmarkUnaryMethod` and `type struct benchmarkStreamingMethod`

we might have a bit of duplicated code from the two structs thoughts. But I believe that future maintenance of the code will be easier with this, we already have a bunch of `if else` in the code, the interface will definitely help for this. 

what do you think?

Reply:
I agree, this makes more sense. Will work on this